libnmstate/nm/connection.py

```diff
@@ -202,6 +205,27 @@ def _active_connection_callback(self, src_object, result, user_data):
                 self._reset_profile()
                 time.sleep(0.1)
                 self.safe_activate_async()
+            elif (
+                # pylint: disable=no-member
+                isinstance(e, nmclient.GLib.GError)
```

Comment:
Instead of `isinstance` it should catch the exception for the generic exception handling:

```
except nmclient.GLib.GError as e:
....
```

Reply:
That is not possible. This except needs to catch all the exceptions, this is why I am using `"isinstance(..)`.

Reply:
Please hide the condition in a method like it was done with `_is_connection_unavailable`, it will make it clear what you are looking for here (by naming it correctly).

Reply:
Done! Thanks